src/api/customers/handlers/get-logged-in-customer.ts

```diff
@@ -21,24 +23,29 @@ export const getLoggedInCustomerQuery = /* GraphQL */ `
     }
   }
 `
-
 export type Customer = NonNullable<GetLoggedInCustomerQuery['customer']>
 
 const getLoggedInCustomer: CustomersHandlers['getLoggedInCustomer'] = async ({
   req,
   res,
   config,
 }) => {
-  const token = req.cookies[config.customerCookie]
+  // use the session if available, otherwise fall back on the customerCookie
+  const session: IronSession = await getIronSession(req, res, sessionOptions)
+  const token: string = session.token || req.cookies[config.customerCookie]
+  const headers: HeadersInit = session.token ? {
+    Authorization: `Bearer ${session.token}`,
+    'X-Bc-Customer-Id': session.customerId.toString()
+  } : {
+    cookie: `${config.customerCookie}=${token}`,
```

Comment:
What is doing this? 🤔

Reply:
So if we have an iron-session, it will use customer impersonation to fulfil the api call, otherwise it falls back to the old cookie-based mechanism.